src/celeritas/optical/action/detail/TrackInitAlgorithms.cu

```diff
@@ -53,18 +69,101 @@ size_type copy_if_vacant(TrackStatusRef<MemSpace::device> const& status,
 
     ScopedProfiling profile_this{"copy-if-vacant"};
 
+    // cub functions expect a cudaStream_t pointer for the stream
+    using StreamT = CELER_DEVICE_API_SYMBOL(Stream_t);
+    StreamT stream = device().stream(stream_id).get();
+
+    // *** For testing with existing code for consistency
+    DeviceVector<size_type> num_vacancies{1, stream_id};
+    // *** For testing with existing code for consistency
+
+    // Calling with nullptr causes the function to return the amount of working
+    // space needed instead of invoking the kernel.
+    size_t temp_storage_bytes = 0;
+#if CELERITAS_USE_CUDA
     auto start = thrust::make_transform_iterator(
         thrust::make_counting_iterator<size_type>(0), TransformType{});
+    auto flags = device_pointer_cast(status.data());
     auto result = device_pointer_cast(vacancies.data());
-    auto end = thrust::copy_if(thrust_execute_on(stream_id),
-                               start,
-                               start + vacancies.size(),
-                               device_pointer_cast(status.data()),
-                               result,
-                               IsVacant{});
-    CELER_DEVICE_API_CALL(PeekAtLastError());
+    DeviceSelect::FlaggedIf(nullptr,
+                            temp_storage_bytes,
+                            start,
+                            flags,
+                            result,
+                            num_vacancies.data(),
```

Comment:
For the "sizing" kernel, do the data pointers have to be valid or can they be nullptr too?

Reply:
Maybe they can probably be nullptr for some. For instance, for the output variable like what's on line 93. For others, I think it may need an actual variable as the working space could be data-type dependent. I've never seen any cub code where they used different variables, though.